core/src/test/java/de/jplag/reporting/jsonfactory/DirectoryManagerTest.java

```diff
@@ -0,0 +1,51 @@
+package de.jplag.reporting.jsonfactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import de.jplag.TestBase;
+
+class DirectoryManagerTest extends TestBase {
+    final String rootDirA = "A";
+    final String fileNameSubmission1AsRootDir = "Submission1.java";
+    final String emptySubDir = "";
+    final String multiSubDirBA = Path.of("B", "A").toString();
+    final String fileNameTerrainType = "TerrainType.java";
+    final String fileNameSubmission1 = "Submission1.java";
+    final String path = Path.of(BASE_PATH, "output", "submissions").toString();
+    final String basecode = "basecode";
+    final String FilesAsSubmissions = "FilesAsSubmissions";
+    final String basecodeWithSameNameOfSubdirectoryAndRootdirectory = "basecode-sameNameOfSubdirectoryAndRootdirectory";
+
+    String[] submissionDirectory = {basecode, FilesAsSubmissions, basecodeWithSameNameOfSubdirectoryAndRootdirectory};
+    String[] names = {rootDirA, fileNameSubmission1AsRootDir, rootDirA};
+    String[] subDirs = {emptySubDir, emptySubDir, multiSubDirBA};
+    String[] fileNames = {fileNameTerrainType, fileNameSubmission1, fileNameTerrainType};
+    File[] files = {new File(Path.of(BASE_PATH, submissionDirectory[0], names[0], subDirs[0], fileNames[0]).toString()),
+            new File(Path.of(BASE_PATH, submissionDirectory[1], subDirs[1], fileNames[1]).toString()),
+            new File(Path.of(BASE_PATH, submissionDirectory[2], names[2], subDirs[2], fileNames[2]).toString())};
+    File[] submissionRoots = new File[files.length];
+    File[] expectedFiles = new File[files.length];
```

Comment:
This is unreadable. The old version was better than this one.

- Constants are written in Java like this `static final String NAME_OF_CONSTANT  = "..."`
- Constants can also be FIles, Paths, etc., and thus can contain String literals
- Assembling constants from constants in an excessive way decreases readability
- A test case should be readable, meaning another person should know immediately what the inputs are, in your case, this is obfuscated!

Reply:
Why do you need the file names in the first place? Can't you just start with a root directory and then get the file names dynamically with `File.list()`?